src/plone/app/multilingual/browser/interfaces.py

```diff
@@ -1,26 +1,28 @@
-from Acquisition import aq_parent
 from plone.app.multilingual import _
 from plone.app.multilingual.browser.vocabularies import untranslated_languages
+from plone.app.multilingual.interfaces import ITranslationManager
 from plone.app.z3cform.widget import RelatedItemsFieldWidget
 from plone.autoform import directives
 from plone.autoform.interfaces import IFormFieldProvider
-from plone.base.interfaces import IPloneSiteRoot
 from plone.supermodel import model
 from z3c.relationfield.schema import RelationChoice
 from zope import interface
 from zope import schema
 from zope.browsermenu.interfaces import IBrowserMenu
 from zope.browsermenu.interfaces import IBrowserSubMenuItem
-from zope.component.hooks import getSite
 from zope.interface import provider
 from zope.schema.interfaces import IContextAwareDefaultFactory
 
 
 def make_relation_root_path(context):
-    ctx = getSite()
-    if not IPloneSiteRoot.providedBy(ctx):
-        ctx = aq_parent(ctx)
-    return "/".join(ctx.getPhysicalPath())
+    current_object = context
+    target_language = context.REQUEST.get("language")
+    closest_translation = None
+    while closest_translation is None:
+        current_object = current_object.aq_parent
+        tm = ITranslationManager(current_object)
+        closest_translation = tm.get_translation(target_language)
```

Comment:
What happens if there's no closest translation, is this even possible? IMO a safety belt like checking and breaking at  IPloneSiteRoot.provided(closest_translation) would be good.
Another case: It could be possible that the closest translation has no view permission. Like there are two teams of editors  for each of en/de, and the other language is not published not have the other team access. Or for whatever reason. Another safety belt: check both, current_object and closest_translation for view permission.

Reply:
Absolutely. Will implement.